### ISSUE TITLE
`bun build` segfaults when using string interpolation in macro arguments.

### ISSUE THREAD

**Issue by Dev-Siri** [oven-sh/bun#6704]
### What version of Bun is running?

1.0.7+b0393fba6200d8573f3433fb0af258a0e33ac157

### What platform is your computer?

Linux 6.2.0-35-generic x86_64 x86_64

### What steps can reproduce the bug?

Create a file like `main.ts` and import a macro

```ts
import { macro } from "./macro" assert { type: "macro" };

const llo = "llo";

// This needs to be a template literal that interpolates a string
console.log(macro(`He${llo}`));
```

The macro is simple:

```ts
export function macro(text: string) {
  return text + " :D";
}
```

Then build it

```sh
$ bun build ./main.ts
```

### What is the expected behavior?

I'd expect this to just run the macro and keep the string interpolation as-is or maybe it will interpolate it in the argument before the macro even runs.
And if its not allowed, (because macros run at build time and string interpolation is done at runtime) then I'd expect to see a friendly error-message rather than it crashing with a segfault.

### What do you see instead?

```sh
[1]    11703 segmentation fault  bun build ./main.ts
```

**Comment by Electroid** [comment]
Related #3830

**Comment by Electroid** [comment]
Closed by #6756
